libs/near-indexer/src/near-indexer.service.ts

```diff
@@ -479,4 +449,45 @@ export class NearIndexerService {
 
     return queryBuilder;
   }
+
+  async findAccountChangeActionsByContractName(
+    contractName: string,
+    fromBlockTimestamp?: number,
+  ): Promise<AccountChange[]> {
+    return this.buildAccountChangeActionQuery(
+      contractName,
+      fromBlockTimestamp,
+    ).getMany();
+  }
+
+  private buildAccountChangeActionQuery(
+    contractName: string,
+    fromBlockTimestamp?: number,
+  ): SelectQueryBuilder<AccountChange> {
+    return this.accountChangeRepository
+      .createQueryBuilder('account_change')
+      .leftJoinAndSelect('account_change.causedByReceipt', 'receipts')
+      .leftJoinAndSelect('receipts.originatedFromTransaction', 'transactions')
+      .leftJoinAndSelect(
```

Comment:
query is too heavy, do we realy need transaction details here?

Reply:
yes we need transaction details to handle the transaction